src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -364,13 +366,14 @@ void QgsAllLayersFeaturesLocatorFilter::prepare( const QString &string, const Qg
                                    string ) );
     req.setLimit( 30 );
 
-    PreparedLayer preparedLayer;
-    preparedLayer.expression = expression;
-    preparedLayer.context = context;
-    preparedLayer.layerId = layer->id();
-    preparedLayer.layerName = layer->name();
-    preparedLayer.iterator =  layer->getFeatures( req );
-    preparedLayer.layerIcon = QgsMapLayerModel::iconForLayer( layer );
+    PreparedLayer *preparedLayer = new PreparedLayer();
```

Comment:
This leaks, right?

Reply:
Either make a `QList<std::shared_ptr<PreparedLayer> > mPreparedLayers;` or a `std::shared_ptr<QgsVectorLayerFeatureSource>`